src/Integrations/Integrations/AMQP/AMQPIntegration.php

```diff
@@ -0,0 +1,409 @@
+            }
+        );
+
+        trace_method(
+            'PhpAmqpLib\Channel\AMQPChannel',
+            'basic_nack',
+            function (SpanData $span, $args, $retval, $exception) use ($integration) {
+                $integration->setGenericTags($span, 'amqp.basic.nack', 'process', $exception);
+
+                /** @var int $deliveryTag */
+                $deliveryTag = $args[0];
+
+                $span->resource = "basic.nack $deliveryTag";
+            }
+        );
+
+        trace_method(
+            'PhpAmqpLib\Channel\AMQPChannel',
+            'basic_get',
+            function (SpanData $span, $args, $message, $exception) use ($integration) {
+                $integration->setGenericTags($span, 'amqp.basic.get', 'consumer', $exception);
+
+                /** @var string $queue */
+                $queue = $args[0];
+
+                $queueDisplayName = $integration->formatQueueName($queue);
+
+                $span->resource = "basic.get $queueDisplayName";
+                $span->meta[Tag::MQ_OPERATION] = 'receive';
+                $span->meta[Tag::MQ_DESTINATION] = $queueDisplayName;
+
+                if (!is_null($message)) {
+                    /** @var AMQPMessage $message */
+                    $exchange = $message->getExchange();
+                    $routingKey = $message->getRoutingKey();
+
+                    $exchangeDisplayName = empty($exchange) ? '<default>' : $exchange;
+                    $routingKeyDisplayName = empty($routingKey)
+                        ? '<all>'
+                        : (str_starts_with($routingKey, 'amq.gen-')
+                            ? '<generated>'
+                            : $routingKey);
+
+                    $span->meta[Tag::MQ_MESSAGE_PAYLOAD_SIZE] = $message->getBodySize();
+
+                    $span->meta[Tag::RABBITMQ_ROUTING_KEY] = $routingKeyDisplayName;
+                    $span->meta[Tag::RABBITMQ_EXCHANGE] = $exchangeDisplayName;
+
+                    $integration->setOptionalMessageTags($span, $message);
+
+                    // Try to extract propagated context values from headers
+                    $integration->extractContext($message);
+                }
+            }
+        );
+
+        return Integration::LOADED;
+    }
+
+    public function formatQueueName($queue)
+    {
+        return empty($queue) || !str_starts_with($queue, 'amq.gen-')
+            ? $queue
+            : '<generated>';
+    }
+
+    public function formatExchangeName($exchange)
+    {
+        return empty($exchange) ? '<default>' : $exchange;
+    }
+
+    public function formatRoutingKey($routingKey)
+    {
+        return empty($routingKey)
+            ? '<all>'
+            : (str_starts_with($routingKey, 'amq.gen-')
+                ? '<generated>'
+                : $routingKey);
+    }
+
+    public function setGenericTags(SpanData $span, string $name, string $spanKind, $exception = null)
+    {
+        $span->name = $name;
```

Comment:
```suggestion
    public function setGenericTags(SpanData $span, string $name, string $spanKind, string $resourceDetail = null, $exception = null)
    {
        $span->name = "amqp.$name";
        $span->resource = "$name" . ($resourceDetail === null ? "" : " $resourceDetail");
```

Wouldn't that reduce redundancy?